Odata-docs/webapiauth/getting-started.md

```diff
@@ -0,0 +1,474 @@
+        {
+            await HttpContext.SignOutAsync(
+                CookieAuthenticationDefaults.AuthenticationScheme);
+
+            return Ok();
+        }
+    }
+}
+```
+
+To test our login flow, we make the following POST request to the login endpoint using a tool like Postman:
+
+```
+POST /auth/login
+
+{
+   "RequestedScopes": ["Product.Read", "Product.Delete"]
+}
+```
+This will create the auth cookie.
+
+To log out, we make the following POST request without a body:
+```
+POST /auth/logout
+```
+
+Now that we have authentication set up, we can finally set up authorization.
+
+## Adding authorization
+
+The authorization middleware will compare the scopes that the authenticated user has to the ones required for the current request based on the modelrestriction annotations. Since there are many ways in which the scopes could be stored, we need to tell the middleware how to extract the scopes from the current user.
+
+Modify `ConfigureServices` to match the following:
+
+```c#
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddDbContext<AppDbContext>(opt => opt.UseInMemoryDatabase("ODataAuthDemo"));
+
+    services.AddOData()
+        .AddAuthorization(options =>
+        {
+            options.ScopesFinder = context =>
+            {
+                var userScopes = context.User.FindAll("Scope").Select(claim => claim.Value);
+                return Task.FromResult(userScopes);
+            };
+
+            options.ConfigureAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
+                .AddCookie();
+        });
+
+    services.AddRouting();
+}
+```
+
+In the above code, we add a call to `.AddAuthorization()` to the odata builder in order to add the WebApi Authorization services. We configure it such that it will extract the scopes from the user claims that were added by the authentication system. We do this by providing a handler method to the `options.ScopesFinder` property. We can also configure authentication directly using `options.ConfigureAuthentication`.
+
+We also need to add `app.UseODataAuthorization()` after `app.UseAuthentication()` in the `Configure()` in order to add the middleware:
+
+```c#
+public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+{
+    if (env.IsDevelopment())
+    {
+        app.UseDeveloperExceptionPage();
+    }
+
+    app.UseRouting();
+
+    app.UseAuthentication();
+
+    app.UseODataAuthorization();
+    
+
+    app.UseEndpoints(endpoints =>
+    {
+        endpoints.Expand().Filter().Count().OrderBy();
+        endpoints.MapODataRoute("odata", "odata", AppEdmModel.GetModel());
+    });
+}
+```
+
+
+## Testing
+
+Let's start off by logging in without any scopes:
+
+```
+POST /auth/login
+
+{ "RequestedScopes": [] }
+```
+
+If we attempt any of the CRUD requests to `/odata/Products`, we should got a `403 Forbidden` response because we don't have the required scopes to access these endpoints.
```

Comment:
```suggestion
If we attempt any of the CRUD requests to `/odata/Products`, we should get a `403 Forbidden` response because we don't have the required scopes to access these endpoints.
```

Reply:
Looks like this one is still pending.